backend-tests/tests/test_decommission.py

```diff
@@ -0,0 +1,272 @@
+import testutils.util.crypto
+from testutils.common import User, Device, Tenant, \
+        create_org, create_random_authset, get_device_by_id_data, \
+        change_authset_status, create_user
+
+logging.basicConfig(format='%(asctime)s %(message)s')
+logger = logging.getLogger('test_decomission')
+logger.setLevel(logging.INFO)
+
+@pytest.yield_fixture(scope='function')
+def clean_migrated_mongo(clean_mongo):
+    deviceauth_cli = CliDeviceauth()
+    useradm_cli = CliUseradm()
+
+    deviceauth_cli.migrate()
+    useradm_cli.migrate()
+
+    yield clean_mongo
+
+@pytest.yield_fixture(scope='function')
+def clean_migrated_mongo_mt(clean_mongo):
+    deviceauth_cli = CliDeviceauth()
+    useradm_cli = CliUseradm()
+    tenantadm_cli = CliTenantadm()
+    for t in ['tenant1', 'tenant2']:
+        deviceauth_cli.migrate(t)
+        useradm_cli.migrate(t)
+        tenantadm_cli.migrate()
+
+    yield clean_mongo
+
+@pytest.yield_fixture(scope="function")
+def user(clean_migrated_mongo):
+    yield create_user('user-foo@acme.com', 'correcthorse')
+
+@pytest.yield_fixture(scope="function")
+def devices(clean_migrated_mongo, user):
+    uc = ApiClient(useradm.URL_MGMT)
+    devauthm = ApiClient(deviceauth_v2.URL_MGMT)
+    devauthd = ApiClient(deviceauth.URL_DEVICES)
+
+    r = uc.call('POST',
+                useradm.URL_LOGIN,
+                auth=(user.name, user.pwd))
+    assert r.status_code == 200
+    utoken = r.text
+
+    aset = create_random_authset(devauthd, devauthm, utoken)
+    dev = Device(aset.did, aset.id_data, aset.pubkey)
+
+    devices = []
+
+    for _ in range(2):
+        aset = create_random_authset(devauthd, devauthm, utoken)
+        dev = Device(aset.did, aset.id_data, aset.pubkey)
+        dev.authsets.append(aset)
+        devices.append(dev)
+
+    yield devices
+
+@pytest.yield_fixture(scope="function")
+def tenants(clean_migrated_mongo_mt):
+    tenants = []
+
+    for n in ['tenant1', 'tenant2']:
+        username = "user@"+n+".com"
+        password = "correcthorse"
+        tenants.append(create_org(n, username, password))
+
+    yield tenants
+
+@pytest.fixture(scope="function")
+def tenants_users_devices(tenants, clean_migrated_mongo_mt):
+    uc = ApiClient(useradm.URL_MGMT)
+    devauthm = ApiClient(deviceauth_v2.URL_MGMT)
+    devauthd = ApiClient(deviceauth.URL_DEVICES)
+    for t in tenants:
+        user = t.users[0]
+        r = uc.call('POST',
+                    useradm.URL_LOGIN,
+                    auth=(user.name, user.pwd))
+        assert r.status_code == 200
+        utoken = r.text
+
+        for _ in range(2):
+            aset = create_random_authset(devauthd, devauthm, utoken, t.tenant_token)
+            dev = Device(aset.did, aset.id_data, aset.pubkey, t.tenant_token)
+            dev.authsets.append(aset)
+            t.devices.append(dev)
+
+    yield tenants
+
+class TestDeviceDecomissioningBase:
+    def do_test_ok(self, user, device, tenant_token=None):
```

Comment:
I think this could be just a function: there is no use of `self`

Reply:
I've just followed the convention we have in the backend integration tests

Reply:
Ok, agreed.